packages/core/components/EditorPanel/DataTableEditor.tsx

```diff
@@ -75,7 +75,7 @@ const DataTable: React.FC<DataTableProps> = ({ config, updateField, isDashboard,
           }
         />
       )}
-      <TextField value={config.table.indexLabel} section='table' fieldName='indexLabel' label='Index Column Header' updateField={updateField} />
+      <TextField value={config.table.indexLabel ? config.table.indexLabel : config.xAxis?.dataKey} section='table' fieldName='indexLabel' label='Index Column Header ddddd' updateField={updateField} />

```

Comment:
@Atash3000 please remove the ddddd part here. I'm also wondering if we should use the data key at all. Most data keys that I've seen aren't the most readable and I'm not sure we want to output that to the user. If I remember correctly, the ticket shared that its okay to output `<span aria-hidden>No title</span>` as a backwards compatible option.

Reply:
If data key is not readable user can override it using Index header input under Data Table section.